### PR TITLE
s3: fix s3 concurrent uploader will overwrite error

### LINKED ISSUE #48164: s3 concurrent uploader will overwrite the real error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
io: read/write on closed pipe
```

see the unit test in https://github.com/pingcap/tidb/pull/48163

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #48164

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/ks3.go [hunk 1]
```diff
@@ -673,9 +673,9 @@ func (rs *KS3Storage) Create(ctx context.Context, name string, option *WriterOpt
 		s3Writer.wg.Add(1)
 		go func() {
 			_, err := up.Upload(upParams)
-			err1 := rd.Close()
 			if err != nil {
-				log.Warn("upload to s3 failed", zap.String("filename", name), zap.Error(err), zap.Error(err1))
+				log.Warn("upload to ks3 failed", zap.String("filename", name), zap.Error(err))
+				_ = rd.CloseWithError(err)
```
Comment (D3Hunter) [inline br/pkg/storage/ks3.go:678]: so there's no need to close rd when Upload returns no err?

Comment (lance6716) [reply br/pkg/storage/ks3.go:678]: Yes, I see PipeReader does not release resources in Close.

Comment (D3Hunter) [reply br/pkg/storage/ks3.go:678]: better call it or explain in the comments, looks like a bug

Comment (lance6716) [reply br/pkg/storage/ks3.go:679]: https://github.com/pingcap/tidb/pull/48163/commits/3a0fd86ae8b9d9e0021f2c135bfa45c4568fcbaf

**Review by D3Hunter — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**